EmeraldIOS/Components/Chart/CustomRenderers/EmeraldCustomBarChartRenderer.swift

```diff
@@ -0,0 +1,611 @@
+//
+//  EmeraldCustomBarChartRenderer.swift
+//  EmeraldIOS
+//
+//  Created by Miguel Roncallo on 9/26/19.
+//  Copyright © 2019 Condor Labs. All rights reserved.
+//
+
+import Foundation
+import Charts
+
+class EmeraldCustomBarChartRenderer: BarChartRenderer {
+    internal var cornerRadius: CGFloat = 0
+    private class Buffer {
+        var rects = [CGRect]()
+    }
+    private var _buffers = [Buffer]()
+    private var _barShadowRectBuffer: CGRect = CGRect()
+    
+    override func initBuffers() {
+        if let barData = dataProvider?.barData
+        {
+            // Matche buffers count to dataset count
+            if _buffers.count != barData.dataSetCount
+            {
+                while _buffers.count < barData.dataSetCount
+                {
+                    _buffers.append(Buffer())
+                }
+                while _buffers.count > barData.dataSetCount
+                {
+                    _buffers.removeLast()
+                }
+            }
+            
+            for i in stride(from: 0, to: barData.dataSetCount, by: 1)
```

Comment:
Taking into account that the stride increasing rate is `1`, don't you like the most using:
```suggestion
            for i in 0..<barData.dataSetCount {
```

Besides! you could use something shorter:
```suggestion
            for set in barData.dataSets where set is IBarChartDataSet {
```

Reply:
Done